### PR TITLE
feat: plugin to produce barker channels

### PR DESCRIPTION
**Pull request by birme**
Plugin `Barker` to produce a barker channel by switching between live channels a given interval

- `BARKERLIST_URL`: URL to a txt file with list of live HLS streams where each line contains a URL to a HLS LIVE stream ([example](https://testcontent.eyevinn.technology/fast/barkertest.txt))
- `BARKER_CHANNEL_NAME`: The name of the channel (default `barker`)
- `SWITCH_INTERVAL_SEC`: How often it should switch between the streams (default 60 sec)

### CODE REVIEW

**Review by Nfrederiksen — APPROVED** [review]
very cool. LGTM, but one question

File: src/plugins/plugin_barker.ts [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+    if (this.liveStreams.length === 0) {
+      console.log("Fetching live stream list from " + this.liveStreamListUrl.toString());
+      const response = await fetch(this.liveStreamListUrl.toString());
+      if (response.ok) {
+        const body = await response.text();
+        this.liveStreams = body.split(/\r?\n/).filter(l => l !== '').map(l => new URL(l.trim()));
+      }  
+    }
+
+    const streamDuration = 60 * 1000;
```
Comment (Nfrederiksen) [inline src/plugins/plugin_barker.ts]: Wait. is this not determined by this.switchIntervalMs value?

**Review by Nfrederiksen — CHANGES_REQUESTED** [review]

File: src/plugins/plugin_barker.ts (hunk 1 above)
Comment (birme) [reply src/plugins/plugin_barker.ts]: You are correct... I'll fix, thanks

**Review by Nfrederiksen — APPROVED** [review]
LGTM for real

**Pull request merged**